### PR TITLE
Add tests and DTO for List Messages in Thread

### PR DESCRIPTION
**Pull request by dominika-kokowicz**
Add tests and DTO for List Messages in Thread

### CODE REVIEW

File: App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListMessagesInThreadRequest.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import Onion
+
+struct ListMessagesInThreadRequest: Request {
+    typealias Output = [Message]
+    
+    var path: String {
+        return preparePathWithComponents()
+    }
+    
+   func preparePathWithComponents() -> String {
```
Comment (sloik) [inline App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListMessagesInThreadRequest.swift:15]: I would like to see more tests for this method :)

File: App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListMessagesInThreadRequest.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import HTTPTypes
+
+// local
+import Onion
+
+struct ListMessagesInThreadRequest: Request {
+    typealias Output = [Message]
+    
+    var path: String {
+        return preparePathWithComponents()
```
Comment (sloik) [inline App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListMessagesInThreadRequest.swift:12]: hint: you can skip the `return`

File: App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListMessagesInThreadRequest.swift [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        if let before = before {
+            components.queryItems?.append(URLQueryItem(name: "before", value: before))
+        }
+        
+        if let after = after {
+            components.queryItems?.append(URLQueryItem(name: "after", value: after))
+        }
+        
+        if components.queryItems != [], let componentsURL = components.url?.absoluteString {
+            return componentsURL
```
Comment (sloik) [inline App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListMessagesInThreadRequest.swift:37]: What will be the `threadId` in here?

File: App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListMessagesInThreadRequest.swift [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+   func preparePathWithComponents() -> String {
+        var components = URLComponents()
+        components.path = self.path
+        components.queryItems = []
+        
+        if let limit = limit {
+            components.queryItems?.append(URLQueryItem(name: "limit", value: String(limit)))
+        }
+        
+        if let offset = offset {
```
Comment (sloik) [inline App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListMessagesInThreadRequest.swift:24]: There is this package https://github.com/sloik/SweetURL/blob/main/Sources/SweetURL/URLExtensions.swift#L59 with method `queryValues`. Could you extension method (maybe to `URLComponents?`) so this call site would look less dense?

**Review by sloik — COMMENTED** [review]
Left so minor suggestions and one important one :D

File: App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListMessagesInThreadRequest.swift [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    var path: String {
+        return preparePathWithComponents()
+    }
+    
+   func preparePathWithComponents() -> String {
+        var components = URLComponents()
+        components.path = self.path
+        components.queryItems = []
+        
+        if let limit = limit {
```
[the remark this answers is not in the document]
Comment (dominika-kokowicz) [reply App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListMessagesInThreadRequest.swift]: Done, thanks

File: App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListMessagesInThreadRequest.swift (hunk 2 above)
Comment (dominika-kokowicz) [reply App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListMessagesInThreadRequest.swift]: Fixed

File: App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListMessagesInThreadRequest.swift (hunk 3 above)
Comment (dominika-kokowicz) [reply App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListMessagesInThreadRequest.swift:34]: Fixed

File: App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListMessagesInThreadRequest.swift (hunk 4 above)
Comment (dominika-kokowicz) [reply App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListMessagesInThreadRequest.swift]: Fixed

**Review by sloik — APPROVED** [review]
Perfect 🎖️

**Pull request merged**